components/arm.py

```diff
@@ -83,76 +104,137 @@ def __init__(self) -> None:
         self.extension_controller = ProfiledPIDController(
-            15, 0, 0, TrapezoidProfile.Constraints(maxVelocity=2, maxAcceleration=2)
+            48, 0, 0, TrapezoidProfile.Constraints(maxVelocity=3.0, maxAcceleration=6)
         )
-        self.extension_simple_ff = SimpleMotorFeedforwardMeters(0, 0.1, 0.1)
+        self.extension_simple_ff = SimpleMotorFeedforwardMeters(kS=0, kV=2, kA=2)
 
         self.brake_solenoid = Solenoid(
             PneumaticsModuleType.CTREPCM, PcmChannels.arm_brake
         )
 
-    def execute(self) -> None:
-        # Clamp extension to not break max height rules
-        is_angle_forwards = math.copysign(self.get_angle() + math.pi / 2, 1)
-        is_goal_forwards = math.copysign(self.goal_angle + math.pi / 2, 1)
-        # if we plan on rotating overhead
-        is_going_over = is_angle_forwards != is_goal_forwards
-        # if we are currently overhead
-        is_currently_up = (
-            self.UPRIGHT_ANGLE > (self.get_angle() + math.pi / 2) > -self.UPRIGHT_ANGLE
+        self.chooser = SendableChooser()
+        self.chooser.addOption("Score Cone High", Setpoints.SCORE_CONE_HIGH)
+        self.chooser.addOption("Score Cube High", Setpoints.SCORE_CUBE_HIGH)
+        self.chooser.addOption("Score Cone Mid", Setpoints.SCORE_CONE_MID)
+        self.chooser.addOption("Score Cube mid", Setpoints.SCORE_CUBE_MID)
+        self.chooser.addOption("Handoff", Setpoints.HANDOFF)
+        self.chooser.setDefaultOption("Handoff", Setpoints.HANDOFF)
+        SmartDashboard.putData(self.chooser)
+        self.last_selection = Setpoint(0, 0)
+
+        # Create arm display
+        self.arm_mech2d = wpilib.Mechanism2d(5, 3)
+        arm_pivot = self.arm_mech2d.getRoot("ArmPivot", 2.5, self.HEIGHT)
+        arm_pivot.appendLigament("ArmTower", self.HEIGHT - 0.05, -90)
+        self.arm_ligament = arm_pivot.appendLigament("Arm", self.MIN_EXTENSION, 0)
+        self.arm_goal_ligament = arm_pivot.appendLigament(
+            "Arm_goal",
+            self.MIN_EXTENSION,
+            0,
+            3,
+            wpilib.Color8Bit(117, 68, 26),
         )
-        should_retract = (is_going_over or is_currently_up) and self.do_auto_retract
-        actual_extension_goal = (
-            self.MIN_EXTENSION if should_retract else self.goal_extension
+        self.arm_extend_ligament = self.arm_ligament.appendLigament(
+            "ArmExtend",
+            0.5,
+            0,
+            8,
+            wpilib.Color8Bit(137, 235, 52),
         )
+        self.arm_extend_goal_ligament = self.arm_ligament.appendLigament(
+            "ArmExtend_goal",
+            0.1,
+            0,
+            4,
+            wpilib.Color8Bit(68, 117, 26),
+        )
+        wpilib.SmartDashboard.putData("Arm sim", self.arm_mech2d)
+
+    def setup(self) -> None:
+        self.set_setpoint(Setpoints.SCORE_CONE_HIGH)
 
+    def execute(self) -> None:
+        setpoint = typing.cast(Setpoint, self.chooser.getSelected())
```

Comment:
😅 the type hints ought to allow this without casting. Could you please file an issue?

In the scenario where we're in the real robot control loop, it'd be preferable to type hint the variable and ignore the type error instead.